File: backend/src/main/java/com/team01/backend/domain/category/dto/CategoryResponseDto.java
package com.team01.backend.domain.category.dto;

import com.team01.backend.domain.category.entity.Category;

public record CategoryResponseDto(
        long id,
        long boardId,
        String name
) {
    public CategoryResponseDto(Category category){
        this(
            category.getId(),
            category.getBoardId(),
            category.getName()
        );
    }
}

File: backend/src/main/java/com/team01/backend/domain/category/dto/CategoryResponseDto.kt
package com.team01.backend.domain.category.dto

import com.team01.backend.domain.category.entity.Category

data class CategoryResponseDto(
    val id: Long,
    val boardId: Long,
    val name: String
) {
    constructor(category: Category) : this(
        category.id,
        category.boardId,
        category.name
    )
}
